website/source/docs/templates/engine.html.md

```diff
@@ -62,6 +62,29 @@ Here is a full list of the available functions for reference.
     each function will behave.
 -   `env` - Returns environment variables. See example in [using home
     variable](/docs/templates/user-variables.html#using-home-variable)
+-   `generated` - This engine will allow you to access special variables that
+    provide connection information and basic instance state information.
+    Usage example:
+    ```json
+        {
+          "type": "shell-local",
+          "environment_vars": ["TESTVAR={{ generated `PACKER_RUN_UUID`}}"],
+          "inline": ["echo $TESTVAR"]
+        },
+    ```
+    Valid variables to request are: "ID", "Host", "Port", "User", "Password",
+    "ConnType", "PACKER_RUN_UUID", "SSHPublicKey", "SSHPrivateKey", and
+    "WinRMPassword", though obviously depending on which communicator you are
+    using, some of these values may be empty. ID represents the vm being
+    provisioned. For example, in Amazon it is the instance id; in digitalocean,
+    it is the droplet id; in Vmware, it is the vm name.
+
+    This function is only for use within _provisioners_, and does not yet work
+    if the provisioners are being used in conjunction with our chroot builders
+    or with lxc/lxd builders.
+
+    This engine is in beta; please report any issues or requests on the Packer
+    issue tracker on GitHub.
```

Comment:
Nice one; 

* I was thinking we could give accessors that are bit more contextual. I gave it some thought and my fav option would be `builder`:
  * ```{{ builder `uuid`}}``` because it is sort of the uuid of the builder
  * as these are for the communicator of the builder: ```{{ builder `communicator_host`}}``` ```{{ builder `communicator_port`}}``` ```{{ builder `communicator_password`}}``` ```{{ builder `communicator_user`}}``` ( we could drop `communicator_winrmpassword` ) ```{{ builder `communicator_pubkey`}}``` ```{{ builder `communicator_privkey`}}``` ```{{ builder `communicator_conn_type`}}```
  * ```{{ builder `identifier`}}``` as it identifies the builder
* do we want to remove the `obviously` ?

Reply:
oh, `build` is a great idea. I love these naming suggestions.

Reply:
`{{ builder `identifier`}} as it identifies the builder` -- not clear on what this would actually be. We have a UUID already; The user can already access the builder type via env vars where relevant. 

There are already places in the config where Host, Port, HTTPIP, etc, are available -- I think having communicator_host is unnecessarily long-winded.

Reply:
I do like changing "generated" to "build", though

Reply:
Oh yes; sorry I first went for builder then changed to build forgot about the last example, I meant: ```{{ build `identifier`}} as it identifies the build```